nni/retiarii/execution/cgo_engine.py

```diff
@@ -143,8 +148,15 @@ def trial_execute_graph(cls) -> None:
 
 class AssemblePolicy:
     @staticmethod
-    def group(logical_plan):
+    def group(logical_plan, available_devices):
+        #TODO: Packing multiple model in one GPU
+        # Currently, we only support one model per GPU
+        all_grouped_models = []
         group_model = {}
+        assert(len(available_devices) > 0) # There should be at least 1 device, set in CGO_DEVICES
         for idx, m in enumerate(logical_plan.models):
-            group_model[m] = PhysicalDevice('server', f'cuda:{idx}')
-        return [group_model]
+            group_model[m] = PhysicalDevice('server', available_devices[idx % len(available_devices)])
+            if len(group_model) == len(available_devices) or idx == len(logical_plan.models)-1:
+                all_grouped_models.append(group_model)
+                group_model = {}
+        return all_grouped_models
```

Comment:
seems this logic is not correct. first, it can only be applied on local mode with a single machine. second, even for a single machine, the gpu resource cannot be fully utilized, for example, there are 4 gpus, logical_plan has 3 gpu, you assign 0/1/2 for them, then a new model comes, you will assign No. 0 gpu, right?

Reply:
You are right. I assumed the models arrive and execute in batch. So the new model should arrive after the previous model finishes.

I will fix it to remove this assumption, so that two models in different batch can be executed together.